### ISSUE TITLE
Craft 3.4.x doesn't support super-table

### ISSUE THREAD

**Issue by ilicmarko** [engram-design/SuperTable#326]
### Description
Updated from `3.4.0-RC3`. Ran migrate/all and project-config/sync. When I go and try to save an entry with a super table field it breaks.

```
yii\base\ErrorException: Invalid argument supplied for foreach() in /webapps/site-admin/current/vendor/verbb/super-table/src/fields/SuperTableField.php:1133
Stack trace:
#0 /webapps/site-admin/current/vendor/craftcms/cms/src/web/ErrorHandler.php(74): yii\base\ErrorHandler->handleError()
#1 /webapps/site-admin/current/vendor/verbb/super-table/src/fields/SuperTableField.php(1133): craft\web\ErrorHandler->handleError()
#2 /webapps/site-admin/current/vendor/verbb/super-table/src/fields/SuperTableField.php(509): verbb\supertable\fields\SuperTableField->_createBlocksFromSerializedData()
#3 /webapps/site-admin/current/vendor/craftcms/cms/src/base/Element.php(2675): verbb\supertable\fields\SuperTableField->normalizeValue()
#4 /webapps/site-admin/current/vendor/craftcms/cms/src/base/Element.php(2279): craft\base\Element->normalizeFieldValue()
#5 /webapps/site-admin/current/vendor/craftcms/cms/src/controllers/EntriesController.php(704): craft\base\Element->setFieldValuesFromRequest()
#6 /webapps/site-admin/current/vendor/craftcms/cms/src/controllers/EntriesController.php(349): craft\controllers\EntriesController->_populateEntryModel()
#7 [internal function]: craft\controllers\EntriesController->actionSaveEntry()
#8 /webapps/site-admin/current/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array()
#9 /webapps/site-admin/current/vendor/yiisoft/yii2/base/Controller.php(157): yii\base\InlineAction->runWithParams()
#10 /webapps/site-admin/current/vendor/craftcms/cms/src/web/Controller.php(178): yii\base\Controller->runAction()
#11 /webapps/site-admin/current/vendor/yiisoft/yii2/base/Module.php(528): craft\web\Controller->runAction()
#12 /webapps/site-admin/current/vendor/craftcms/cms/src/web/Application.php(290): yii\base\Module->runAction()
#13 /webapps/site-admin/current/vendor/craftcms/cms/src/web/Application.php(558): craft\web\Application->runAction()
#14 /webapps/site-admin/current/vendor/craftcms/cms/src/web/Application.php(269): craft\web\Application->_processActionRequest()
#15 /webapps/site-admin/current/vendor/yiisoft/yii2/base/Application.php(386): craft\web\Application->handleRequest()
#16 /webapps/site-admin/current/web/index.php(21): yii\base\Application->run()
#17 {main}
```

After some investigation, I found that `$value['sortOrder']` is a string. For this reason on line 1124

https://github.com/verbb/super-table/blob/0c42d9f41a984678569fb030c387754726769a84/src/fields/SuperTableField.php#L1124

It will assign String value.

Here is the data sent from Craft:

```
'gridblock' => [
            'sortOrder' => 'new1',
            'blocks' => [
                'new1' => [
                    'type' => '16',
                    'fields' => [
                        'header' => 'Cao',
                        'text' => '<p>Test</p>',
                    ],
                ],
            ],
        ],
```

### Steps to reproduce

1. Create a super table field
2. Attach it to a section
3. Try to save

### Application Info

|         Info                  |            Version             |
| ------------------------- | ----------------------- |
| PHP version               | 7.4.2                   |
| OS version                | Linux 4.15.0-74-generic |
| Database driver & version | MySQL 5.7.29            |
| Image driver & version    | GD 7.4.2                |
| Craft edition & version   | Craft Pro 3.4.0.2       |
| Yii version               | 2.0.32                  |
| Twig version              | 2.12.3                  |
| Guzzle version            | 6.5.1                   |
| Imagine version           | 1.2.3-dev               |

## Plugins

|       Plugin        |    Version     |
| ------------- | ------- |
| Communicator  | 1.0.0   |
| Feed Me       | 4.2.0.1 |
| Field Manager | 2.1.1   |
| Guide         | 2.1.3.2 |
| Redactor      | 2.5.0   |
| SEO           | 3.6.3   |
| Super Table   | 2.4.0   |
| Webhooks      | 2.2.0   |

**Comment by MarcHartwig13** [comment]
Same issue here. Updating from 3.4.0-RC3, also updating from SuperTable 2.3.3. 

### Description

When I clicked to add a new Super Table row in an entry I get `A server error occurred.` there is also a red flag next to the entry version with the message `The draft could not be saved.` An error pops up in the console with `uncaught exception: undefined` (no stack trace or file with line numbers shows up)

When I hit save I get I also get the `foreach` error:
`Invalid argument supplied for foreach()`

```
2020-01-29 09:29:48 [-][1][di7s7pj4fagc48c9agkmr34ot9][error][yii\base\ErrorException:2] yii\base\ErrorException: Invalid argument supplied for foreach() in /website/vendor/verbb/super-table/src/fields/SuperTableField.php:1133
Stack trace:
#0 /website/vendor/craftcms/cms/src/web/ErrorHandler.php(74): yii\base\ErrorHandler->handleError(2, 'Invalid argumen...', '/Users/marc_har...', 1133)
#1 /website/vendor/verbb/super-table/src/fields/SuperTableField.php(1133): craft\web\ErrorHandler->handleError(2, 'Invalid argumen...', '/Users/marc_har...', 1133, Array)
#2 /website/vendor/verbb/super-table/src/fields/SuperTableField.php(509): verbb\supertable\fields\SuperTableField->_createBlocksFromSerializedData(Array, Object(craft\elements\MatrixBlock))
#3 /website/vendor/craftcms/cms/src/base/Element.php(2675): verbb\supertable\fields\SuperTableField->normalizeValue(Array, Object(craft\elements\MatrixBlock))
#4 /website/vendor/craftcms/cms/src/base/Element.php(2157): craft\base\Element->normalizeFieldValue('iconCard')
#5 /website/vendor/craftcms/cms/src/base/Element.php(1094): craft\base\Element->getFieldValue('iconCard')
#6 /website/vendor/yiisoft/yii2/validators/Validator.php(303): craft\base\Element->__get('field:iconCard')
#7 /website/vendor/yiisoft/yii2/validators/Validator.php(257): yii\validators\Validator->validateAttribute(Object(craft\elements\MatrixBlock), 'field:iconCard')
#8 /website/vendor/yiisoft/yii2/base/Model.php(367): yii\validators\Validator->validateAttributes(Object(craft\elements\MatrixBlock), Array)
#9 /website/vendor/craftcms/cms/src/fields/Matrix.php(713): yii\base\Model->validate()
#10 /website/vendor/craftcms/cms/src/base/Element.php(1383): craft\fields\Matrix->validateBlocks(Object(craft\elements\Entry), NULL)
#11 [internal function]: craft\base\Element->validateCustomFieldAttribute('field:landingPa...', Array, Object(yii\validators\InlineValidator))
#12 /website/vendor/yiisoft/yii2/validators/InlineValidator.php(72): call_user_func(Array, 'field:landingPa...', Array, Object(yii\validators\InlineValidator))
#13 /website/vendor/yiisoft/yii2/validators/Validator.php(257): yii\validators\InlineValidator->validateAttribute(Object(craft\elements\Entry), 'field:landingPa...')
#14 /website/vendor/yiisoft/yii2/base/Model.php(367): yii\validators\Validator->validateAttributes(Object(craft\elements\Entry), Array)
#15 /website/vendor/craftcms/cms/src/services/Elements.php(1975): yii\base\Model->validate()
#16 /website/vendor/craftcms/cms/src/services/Elements.php(524): craft\services\Elements->_saveElementInternal(Object(craft\elements\Entry), true, true, true)
#17 /website/vendor/craftcms/cms/src/controllers/EntryRevisionsController.php(378): craft\services\Elements->saveElement(Object(craft\elements\Entry))
#18 [internal function]: craft\controllers\EntryRevisionsController->actionPublishDraft()
#19 /website/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#20 /website/vendor/yiisoft/yii2/base/Controller.php(157): yii\base\InlineAction->runWithParams(Array)
#21 /website/vendor/craftcms/cms/src/web/Controller.php(178): yii\base\Controller->runAction('publish-draft', Array)
#22 /website/vendor/yiisoft/yii2/base/Module.php(528): craft\web\Controller->runAction('publish-draft', Array)
#23 /website/vendor/craftcms/cms/src/web/Application.php(290): yii\base\Module->runAction('entry-revisions...', Array)
#24 /website/vendor/craftcms/cms/src/web/Application.php(558): craft\web\Application->runAction('entry-revisions...', Array)
#25 /website/vendor/craftcms/cms/src/web/Application.php(269): craft\web\Application->_processActionRequest(Object(craft\web\Request))
#26 /website/vendor/yiisoft/yii2/base/Application.php(386): craft\web\Application->handleRequest(Object(craft\web\Request))
#27 /website/web/index.php(21): yii\base\Application->run()
#28 /Users/marc_hartwig/.composer/vendor/laravel/valet/server.php(158): require('/Users/marc_har...')
#29 {main}

```


### Steps to reproduce

1. Navigate to entry with a super table
2. Add row
3. Attempt to save

### Additional info

- Plugin version: 2.4.0
- Craft version: 3.4.0.2

**Comment by jamielaughton** [comment]
Exactly the same error here, hope this can be be resolved soon!

**Comment by philipzaengle** [comment]
Same issue here!

**Comment by philipzaengle** [comment]
This appears to be fixed in dc8f499644cbdb0637c95fc88d2bf8fa14aef8e9, but a new release has not yet been tagged.

**Comment by httpster** [comment]
Same issue here. Hopefully they tag the release @philipzaengle mentioned above soon.

**Comment by ilicmarko** [comment]
I didn't see that. I almost made that same PR because that should be the PR, looking at it from the phone 😄

**Comment by engram-design** [comment]
Just have a few more fixes to apply today, then the release will be tagged

**Comment by engram-design** [comment]
But, to get it early, modify your `composer.json` file to `verbb/super-table: "dev-craft3 as 2.4.0.1"`, then run `composer update`

**Comment by engram-design** [comment]
Should be fixed in [2.4.1](https://github.com/verbb/super-table/releases/tag/2.4.1)
